src/main/scala/uk/gov/hmrc/ct/box/Validators.scala

```diff
@@ -52,6 +52,21 @@ trait Validators[B <: BoxRetriever] {
     }
   }
 
+  protected def validate(boxId: String = boxId)(predicate: (B) => Boolean)(errors: Set[CtValidation])(boxRetriever: B): Set[CtValidation] = {
+    if (predicate(boxRetriever))
+      errors
+    else
+      Set.empty
+  }
+
+  protected def validateMoney(boxId: String = boxId)(value: Option[Int], min: Int = -99999999, max: Int = 99999999)(boxRetriever: B): Set[CtValidation] = {
```

Comment:
Not quite. It fits into the framework I built for trading losses. Maybe we need to decide which path to take...